### PR TITLE
Configuration migrations

### PR DESCRIPTION
**Pull request by lbarcziova**
See commit messages and help of the commands for more info

TODO:
- [ ] test the migrate command
- [ ] move this script somewhere else?

### CODE REVIEW

File: automatic-config-updates/packit_config_checker.py [hunk 1]
```diff
@@ -0,0 +1,491 @@
...
+            parsed_url = urlparse(url)
+            sanitized_path = re.sub(r"[^\w_-]", "_", parsed_url.path)
+            dir_name = f"{parsed_url.netloc}{sanitized_path}"
+            dir_path = os.path.join(config_dir, dir_name)
+
+            Path(dir_path).mkdir(parents=True, exist_ok=True)
+
+            file_path = os.path.join(dir_path, repo.package_config_path)
+            with open(file_path, "w") as config_file:
+                config_file.write(repo.package_config)
```
Comment (jpopelka) [inline automatic-config-updates/packit_config_checker.py:362]:
```suggestion
            dir_path = repo_store_path / dir_name

            dir_path.mkdir(parents=True, exist_ok=True)

            file_path = dir_path / repo.package_config_path
            file_path.write_text(repo.package_config)
```

File: automatic-config-updates/packit_config_checker.py [hunk 2]
```diff
@@ -0,0 +1,491 @@
...
+        self._package_config: Optional[str] = None
+
+    @classmethod
+    def get_url_from_file(cls, directory: Path) -> Optional[str]:
+        url_file = os.path.join(directory, URL_FILE_NAME)
+        if not os.path.isfile(url_file):
+            return None
+
+        with open(url_file) as file:
+            url = file.read()
```
Comment (jpopelka) [inline automatic-config-updates/packit_config_checker.py:107]:
```suggestion
        url_file = directory / URL_FILE_NAME
        if not url_file.is_file():
            return None

        url = url_file.read_text()
```

**Review by jpopelka — COMMENTED** [review]
Nicely structured, LGTM

**Review by TomasTomecek — APPROVED** [review]
this is really impressive

would be amazing to move all the code in a dedicated project: I have a sense we'll use it more in the future :)

**Comment by lbarcziova** [comment]
@TomasTomecek yes, agreed, I will move it to a new dedicated repo; so now the harder part, what should be the name :grinning:  :  
- (automatic)-configuration-updates
- (automatic)-configuration-migrations
- configuration-checker
- ?

**Comment by nforro** [comment]
> * (automatic)-configuration-migrations

I think this makes sense.

**Comment by lbarcziova** [comment]
The content of this PR was moved to [the new repo](https://github.com/packit/configuration-migrations), therefore closing.

**Pull request closed without merging**